lib/jekyll/static_file.rb

```diff
@@ -144,7 +144,11 @@ def copy_file(dest_path)
       if @site.safe || Jekyll.env == "production"
         FileUtils.cp(path, dest_path)
       else
-        FileUtils.copy_entry(path, dest_path)
+        begin
+          FileUtils.ln(path, dest_path)
+        rescue
```

Comment:
Let the error surface and throw the application. It's the user and the file systems job to deal with those kinds of problems past making sure the old exists and the new doesn't exist.  We don't validate permissions, too many links, unaddressed links, memory and other stuff that might surface through link into File#link into Ruby's native errors.

As far as checking if link exists, it should exist on all Unix and Unix like machines in one form or another through it's C library, except for Windows... I would simply just check if it's Windows, but at that same point, I'm hesistant to accept a feature that doesn't work across all platforms as we need stop treating Windows like it's second-class.